File: LeetCode/523-continuous-subarray-sum/ex_523.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        if (nums.size() < 2)    return false;
        unordered_set<int> seen;
        seen.insert(0);
        seen.insert(nums[0]);
        int sum = nums[0];
        for (auto i = 1; i != nums.size(); ++i) {
            if (nums[i] == 0 && nums[i-1] == 0)     return true;
            else if (k == 0)    continue;
            else {
                sum = (sum + nums[i]) % k;
                if (seen.find(sum) != seen.end())   return true;
                seen.insert(sum);
            }
            nums[i] = nums[i] % k;
        }
        return false;
    }
};

File: LeetCode/523-continuous-subarray-sum/ex_523.py
class Solution(object):
    def checkSubarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        if len(nums) < 2:   return False
        seen = set([nums[0], 0])
        s = nums[0]
        for i in range(1, len(nums)):
            if nums[i] == nums[i-1] == 0:    return True
            elif k == 0: continue
            else:
                s = (s + nums[i]) % k
                if s in seen:   return True
                seen.add(s)
        return False